sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -783,7 +786,10 @@ __ESIMD_API
                __ESIMD_NS::simd_mask<N> pred = 1) {
 #ifdef __ESIMD_FORCE_STATELESS_MEM
   return lsc_gather<T, NElts, DS, L1H, L2H>(
-      reinterpret_cast<T *>(acc.get_pointer().get()), offsets, pred);
+      reinterpret_cast<std::add_pointer_t<sycl::detail::copy_cv_qualifiers_t<
+          typename AccessorTy::value_type, T>>>(
+          acc.template get_multi_ptr<access::decorated::legacy>().get()),
```

Comment:
Note for reviewers: this is replacing one deprecated API with another deprecated API that is `access::decorated::legacy`. It should be in fact either `decorated::yes`, or `decorated::no`, but I'm not familiar enough with that part of the spec and details of ESIMD implementation to say which one is better here.

Tagging @steffenlarsen to comment and provide feedback here

Reply:
I think what we want to do in ESIMD here is pretty simple. Can you explain what decorated and not decorated means? I/We can probably suggest which one to use.

Reply:
From what I understand, `decorated::no` means that there won't be any implementation-defined address space attributes applied to a pointer, meaning that it is essentially a pointer in `generic` address space. It would be up to backend to figure out the exact address space and generate corresponding code.

If needed/preferred, a pointer can be explicitly decorated with an address space to help backend figure it out. If I understand correctly, the main use case is essentially performance tuning.

Corresponding sections in the SYCL spec: https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_address_space_classes

`legacy` refers to SYCL 1.2.1 handling, which was not so precise, i.e. it was either all generic or all named address spaces depending on compiler flags/configuration. For us it was all generic, so theoretically, `legacy` should be equal to `no` for us.

Reply:
I think this code should use decorated::yes.

The more general/unified fix would be: 
return lsc_gather<T, NElts, DS, L1H, L2H>(
    __ESIMD_DNS::detail::accessorToPointer<T>(acc), offsets, pred);

Then it would require the fix in just one place: [accessorToPointer](https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/intel/esimd/detail/util.hpp#L199), which is used in many other places (e.g. [here ](https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/intel/esimd/memory.hpp#L3421) and [here](https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/intel/esimd/memory.hpp#L7176)) and which is currently using that obsolete get_pointer() method.

Reply:
Thanks for the hint, I will update the PR to use that helper and update the helper to use `get_multi_ptr()` with `decorated::yes`

Reply:
I've made changes according to @v-klochkov suggestions in 229246b1a02f52b6368f35981c6936fd16dc3afa